sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuParseUrl.scala

```diff
@@ -79,7 +79,7 @@ case class GpuParseUrl(children: Seq[Expression])
 
   def doColumnar(col: GpuColumnVector, partToExtract: GpuScalar, key: GpuScalar): ColumnVector = {
     val part = partToExtract.getValue.asInstanceOf[UTF8String].toString
-    if (part != QUERY) {
+    if (part != QUERY || key == null || key.getValue == null) {
```

Comment:
nit: `key.isValid` instead of `key.getValue == null`.

Reply:
Ah, yes, of course.  Will fix.